### PR TITLE
Project sanity check

### PR DESCRIPTION
**Pull request by BytewaveMLP**
- Marginally fixed build system

  - Brought in @LunarMist's tslint changes

Circular dependencies are still an ongoing issue.

### CODE REVIEW

**Comment by BytewaveMLP** [comment]
I know the commit history is a mess—sorry about that. Remember to squash and merge. :P 

This combines both #42 and my intended fixes to the build system and .gitignores into one shiny package.

File: gulpfile.js [hunk 1]
```diff
@@ -323,6 +323,7 @@ const setProd = cb => {
 };
 
 const empty = cb => cb();
+
```
Comment (blackhole12) [inline gulpfile.js]: Since this doesn't actually change anything in gulpfile, maybe just remove this additional newline so it's not in the changelog?

**Comment by BytewaveMLP** [comment]
re: `package-lock.json` and `sprites.ts`

I assume so, Git probably converted crlf to lf for me or something and didn't yell at me.

**Comment by blackhole12** [comment]
Those files should have been committed with LF endings on the initial commit. Can you double check to see if they have CRLF in the actual repo? If they do then the line ending fix is inevitable but otherwise we should avoid committing massive changes like that.

**Comment by BytewaveMLP** [comment]
Strange... looks like the opposite happened. They went from LF to CRLF despite editorconfig and tslint not allowing either.

Let me fix that.

**Comment by Stubenhocker1399** [comment]
With removal of `pony.bin` building the project now initially requires building with `--sprites`. Maybe update the readme to reflect this?

**Comment by BytewaveMLP** [comment]
That's a good point actually. Not sure why that didn't come up in my testing.

I'll look into that when I get home from class. I think my original plan was to allow sprites to be built outside of `dev` mode, but that requires a TS build, which requires a sprite build, which requires a build, which requires a TS build... ugh. Fortunately, again, we provide an old copy of `sprites.ts`.

**Comment by BytewaveMLP** [comment]
@Stubenhocker1399 Added the ability to build sprites without a development build (technically those are faster, but it's nice to have a one-stop shop). The README note may be a little too small... hopefully people pick up on it, though.

**Pull request merged**